app/jobs/create_daily_summary_job.rb

```diff
@@ -0,0 +1,22 @@
+class CreateDailySummaryJob < ApplicationJob
+  queue_as :low
+
+  def perform(impressionable, date_string, scoped_by)
+    date = Date.coerce(date_string)
+    return if date.today? || date.future?
+    return if impressionable.daily_summaries.on(date).scoped_by(scoped_by).exists?
+
+    impressions = impressionable.impressions.on(date).scoped_by(scoped_by)
+
+    impressionable.daily_summaries.create!(
+      displayed_at_date: date,
+      scoped_by_type: scoped_by&.class&.name,
+      scoped_by_id: scoped_by&.id,
+      impressions_count: impressions.count,
+      clicks_count: impressions.clicked.count,
```

Comment:
What are your thoughts on adding  CTR and CPC to this as well? It seems like a simple add and it would reduce us from having to calculate it on the fly. Also, perhaps we can add Paid Ad %?

Reply:
We already have `click_rate` and `ecpm`. I like the idea of adding `cpc` will get that in.